### ISSUE TITLE
parallel function reports step failure reason as '{}'

### ISSUE THREAD

**Issue by achingbrain** [cucumber/cucumber-js#1027]
The reason for step failures is being reported as `{}` by the default formatter when using the new `parallel` feature.

I think this may be because a JavaScript `Error` object is being run through `JSON.stringify`, probably as part of the communication between the master and the slave.

The event message received by the master looks like this:

```
{"command":"event","name":"test-step-finished","data":{"index":2,"result":{"duration":90055,"exception":{},"status":"failed"},"testCase":{"sourceLocation":{"uri":"path/to/my.feature","line":7}}}}
```

**Comment by achingbrain** [comment]
This might just be when steps time out as it looks like it's a generic `Error` object that is passed through.  `AssertionError`s get reported correctly:

```
{"command":"event","name":"test-step-finished","data":{"index":6,"result":{"duration":1001,"exception":{"name":"AssertionError","message":"expected 'true' to equal 'false'","showDiff":true,"actual":"true","expected":"false","stack":"AssertionError: expected 'true' to equal 'false'\n    at World.<anonymous>  BIG_STACK_TRACE_HERE"},"status":"failed"},"testCase":{"sourceLocation":{"uri":"path/to/my.feature","line":20}}}}
```

**Comment by charlierudolph** [comment]
Is the error something other than a `{}` when run without the parallel flag?

**Comment by achingbrain** [comment]
Yes, without the parallel flag I see the stack trace as expected:

```
   ✖ Given I do something # ./step-definitions/navigation-steps.js:21
       Error: function timed out after 90000 milliseconds
           at Timeout.<anonymous> (/path/to/node_modules/cucumber/lib/user_code_runner.js:93:22)
           at ontimeout (timers.js:386:11)
           at tryOnTimeout (timers.js:250:5)
           at Timer.listOnTimeout (timers.js:214:5)
```

vs:

```
   ✖ Given I do something # ./step-definitions/navigation-steps.js:21
       {}
```

**Comment by charlierudolph** [comment]
fixed in #1053
